src/System.Drawing.Common/tests/IconTests.cs

```diff
@@ -495,40 +492,96 @@ public void ToBitmap_BitmapIcon_Success(Icon icon)
-                long length = stream.Length - 22;
-                stream.Seek(14, SeekOrigin.Begin);
-                stream.WriteByte((byte)length);
-                stream.WriteByte((byte)(length >> 8));
+                Icon icon;
+                using (var stream = new MemoryStream())
+                {
+                    // Create a PNG inside an ICO.
+                    var bitmap = new Bitmap(10, 10);
+                    stream.Write(new byte[] { 0, 0, 1, 0, 1, 0, (byte)bitmap.Width, (byte)bitmap.Height, 0, 0, 0, 0, 32, 0, 0, 0, 0, 0, 22, 0, 0, 0 }, 0, 22);
+
+                    // Writing actual data
+                    bitmap.Save(stream, ImageFormat.Png);
+
+                    // Getting data length (file length minus header)
+                    long length = stream.Length - 22;
+                    stream.Seek(14, SeekOrigin.Begin);
+                    stream.WriteByte((byte)length);
+                    stream.WriteByte((byte)(length >> 8));
+
+                    // Read the PNG inside an ICO.
+                    stream.Position = 0;
+                    icon = new Icon(stream);
+                }
 
-                // Read the PNG inside an ICO.
-                stream.Position = 0;
-                icon = new Icon(stream);
-            }
+                using (Bitmap bitmap = icon.ToBitmap())
+                {
+                    Assert.NotSame(icon.ToBitmap(), bitmap);
+                    Assert.Equal(PixelFormat.Format32bppArgb, bitmap.PixelFormat);
+                    Assert.Empty(bitmap.Palette.Entries);
+                    Assert.Equal(icon.Width, bitmap.Width);
+                    Assert.Equal(icon.Height, bitmap.Height);
+
+                    Assert.Equal(ImageFormat.Png, bitmap.RawFormat);
+                    Assert.Equal(77842, bitmap.Flags);
+                }
 
-            using (Bitmap bitmap = icon.ToBitmap())
+                return SuccessExitCode;
+            }).Dispose();
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        public void ToBitmap_PngIconNotSupportedInSwitches_ThrowsArgumentOutOfRangeException()
+        {
+            RemoteInvoke(() =>
             {
-                Assert.NotSame(icon.ToBitmap(), bitmap);
-                Assert.Equal(PixelFormat.Format32bppArgb, bitmap.PixelFormat);
-                Assert.Empty(bitmap.Palette.Entries);
-                Assert.Equal(icon.Width, bitmap.Width);
-                Assert.Equal(icon.Height, bitmap.Height);
+                AppContext.SetSwitch("Switch.System.Drawing.DontSupportPngFramesInIcons", true);
 
-                Assert.Equal(ImageFormat.Png, bitmap.RawFormat);
-                Assert.Equal(77842, bitmap.Flags);
-            }
+                Icon icon;
+                using (var stream = new MemoryStream())
+                {
+                    // Create a PNG inside an ICO.
+                    var bitmap = new Bitmap(10, 10);
```

Comment:
All the handles and stuff are disposed during finalization, but I agree. For now I'll only update usings introduced or changed in this PR though and keep that for another PR